### ISSUE TITLE
Icon component: underlying HTML changes for a11y

### ISSUE THREAD

**Issue by jbadan** [SAP/fundamental#1351]
Icons: https://sap.github.io/fundamental/components/icon.html

per "Icons can be used adaptively if desired, but at this point they are used more as visual elements within other components", we need to be able to render both a` <span>` and `<button>`. When rendering a `<span>` a `role='image'` should be added along with `alt='{icon-description}'`

If this already does work, consider adding examples to the documentation site.
